compiler/rustc_trait_selection/src/traits/dyn_compatibility.rs

```diff
@@ -39,6 +39,16 @@ pub fn hir_ty_lowering_dyn_compatibility_violations(
     trait_def_id: DefId,
 ) -> Vec<DynCompatibilityViolation> {
     debug_assert!(tcx.generics_of(trait_def_id).has_self);
+
+    // Check for `AsyncFn` traits first to avoid reporting various other
```

Comment:
This isn't necessary, is it? I would suppose removing this doesn't actually affect anything.